src/Eloquent/Builder.php

```diff
@@ -181,7 +181,10 @@ public function raw($value = null)
 
         // Convert MongoCursor results to a collection of models.
         if ($results instanceof CursorInterface) {
-            $results->setTypeMap(['root' => 'array', 'document' => 'array', 'array' => 'array']);
+            if(!config('database.connections.mongodb.options.AggregateCollectionArray')){
+                $results->setTypeMap(['root' => 'array', 'document' => 'array', 'array' => 'array']);
```

Comment:
If your callback does not return a `CursorInterface`, this behavior will be disabled. Use `Cursor::toArray()` for example.

```php
Model::raw(function (\MongoDB\Collection $collection): array {
    $cursor = $collection->find([], ['typeMap' => ['array' => 'bson']);

    return $cursor->toArray();
});
```